frontend-react/src/pages/admin/value-set-editor/ValueSetsIndex.test.tsx

```diff
@@ -79,4 +92,23 @@ describe("ValueSetsIndex tests", () => {
         ).toBeInTheDocument();
         expect(within(firstContentRow).getByText("you")).toBeInTheDocument();
     });
+    test("Error in query will render error UI instead of table", () => {
+        mockUseValueSetsMeta = jest.fn(
+            () =>
+                ({
+                    valueSetMeta: fakeMeta,
+                } as ValueSetsMetaResponse)
+        );
+        mockUseValueSetsTable = jest.fn(() => {
+            throw new RSNetworkError("Test", 404);
+        });
+        /* Outputs a large error stack...should we consider hiding error stacks in page tests since we
```

Comment:
that is a large stack! for testing purposes I think we're fine since we have the `--silent` flag on. This full log will run in the console if an error actually occurs though, and it's big enough to be unhelpful. 

A brief investigation indicates that the error boundary log is relatively small and helpful. It has a full stack trace that is probably overkill but we could address that later. More interesting to me is that most of the log is not coming from there, but  from the error not being caught somewhere farther up the chain. I wonder if we need to be catching and rethrowing the error at some point to avoid this? I'm not especially worried about it for now,b ut something to keep an eye on

Reply:
So this is a bit of a weird phenomenon. I've noticed that it'll say "uncaught error" even if the boundary catches it. I'll have to ask around some React groups to see what the deal is with that, because so far as I've seen, throwing anywhere from the wrapped component, all the way down into the functions themselves, results in a false "uncaught" tag on the error. 😮

Reply:
This is the best I could find on the matter: https://github.com/facebook/react/issues/15069

It seems intentional, which is kinda annoying! Seems I'm not the only less-than-thrilled user: https://github.com/facebook/react/issues/15069#issuecomment-845081191